AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerServiceImpl.kt

```diff
@@ -162,6 +172,34 @@ internal class TestRunnerServiceImpl internal constructor(
                 getTestResultFiles(visitor).logcat = ResultFileResourceImpl(visitor)
             } else if (fileName == INSTRUMENTATION_RESULTS_FILE_NAME) {
                 getTestResultFiles(visitor).instrumentationResult = ResultFileResourceImpl(visitor)
+            } else if (fileName.endsWith(LOGCAT_FILE_NAME_SUFFIX)) {
+                val step = steps.flatMap {
+                    it.testExecutionStep?.toolExecution?.toolOutputs!!
+                }?.find {
+                    (it.output?.fileUri == visitor.gcsPath.toString())
+                }
+                val fileUri = step?.output?.fileUri
+                var attemptNumber = 0
+                if (fileUri?.contains("rerun_1") == true) {
```

Comment:
you need to parse these properly, there might be N number of reruns. see the code which parses the outputs where we already have the regex's to do that properly.